EngineLayer/Proteomics/Protease.cs

```diff
@@ -54,6 +54,19 @@ public override int GetHashCode()
             return Name.GetHashCode();
         }
 
+        public void IsSemiSpecific(SearchType searchType, TerminusType terminusType)
```

Comment:
Instead of this method, make a new constructor.

Then put this code in SearchTask before  HashSet<DigestionParams> ListOfDigestionParams is created
            if(nonspecific)
            {

                CommonParameters.DigestionParams.Protease = new Protease(CommonParameters.DigestionParams.Protease, terminusType);
                foreach(var digestionParams in fileSettingsList)
                {
                    digestionParams.Protease = new Protease(digestionParams.Protease,  terminusType);

                }
            }